tcti/commonchecks.c

```diff
@@ -38,18 +38,20 @@ TSS2_RC CommonSendChecks(
     uint8_t           *command_buffer     /* in */
     )
 {
+    TSS2_TCTI_CONTEXT_INTEL *tcti_intel = tcti_context_intel_cast (tctiContext);
+
     if( tctiContext == NULL || command_buffer == NULL )
     {
         return TSS2_TCTI_RC_BAD_REFERENCE;
     }
 
-    if (TCTI_CONTEXT_INTEL->previousStage == TCTI_STAGE_SEND_COMMAND)
+    if (tcti_intel->previousStage == TCTI_STAGE_SEND_COMMAND)
     {
         return TSS2_TCTI_RC_BAD_SEQUENCE;
     }
 
-    if (TCTI_CONTEXT_INTEL->magic != TCTI_MAGIC ||
-        TCTI_CONTEXT_INTEL->version != TCTI_VERSION)
+    if (tcti_intel->magic != TCTI_MAGIC ||
+        tcti_intel->version != TCTI_VERSION)
```

Comment:
Call me crazy, but id move these MAGIC and VERSION checks to the cast routine....

Reply:
```
ptr = cast(old)
if (!ptr) {
  return BAD_REF;
}
```

Reply:
in the cast routine:

```
type *cast (old) {
  new = (type *)old;
  if (!new || new->nagic != TCTI_MAGIC || new->VERSION != TCTI_VERSION) {
     return NULL;
  }
  return new;
}
```

Reply:
even better that the current implementation checks the MAGIC and VERSION fields after it accesses private data in the context structure ... whoops. Should probably fix that too.

Reply:
ugh I'm just now realizing why I didn't make this change in the first PR: the TCTI functions are still in one big `if` block. I really want to remove this before making any significant changes to the body of the functions.

@williamcroberts are you OK with merging this PR as-is so that I can restructure these functions before taking on the MAGIC & VERSION checks?

Reply:
Thats fine...